rest-service/manager_rest/rest/resources_v3_1/secrets.py

```diff
@@ -7,18 +7,16 @@
                                          generate_key_using_password)
```

Comment:
Any reason why you've not just copied this over the 3.0 one and putting in the extra decrypt-fail catching there?